src/Verilator.cpp

```diff
@@ -116,474 +119,487 @@ static void reportStatsIfEnabled() {
-        // Check XML when debugging to make sure no missing node types
-        || (v3Global.opt.debugCheck() && !v3Global.opt.lintOnly() && !v3Global.opt.dpiHdrOnly())) {
-        V3EmitXml::emitxml();
-    }
+    {
+        const V3MtDisabledLockGuard mtDisabler{v3MtDisabledLock()};
 
-    // Output DPI protected library files
-    if (!v3Global.opt.libCreate().empty()) {
-        if (v3Global.rootp()->delaySchedulerp()) {
-            v3warn(E_UNSUPPORTED, "Unsupported: --lib-create with --timing and delays");
+        if (v3Global.opt.xmlOnly()
+            // Check XML when debugging to make sure no missing node types
+            || (v3Global.opt.debugCheck() && !v3Global.opt.lintOnly()
+                && !v3Global.opt.dpiHdrOnly())) {
+            V3EmitXml::emitxml();
         }
-        V3ProtectLib::protect();
-        V3EmitV::emitvFiles();
-        V3EmitC::emitcFiles();
-    }
 
-    if (v3Global.opt.stats()) V3Stats::statsStage("emit");
+        // Output DPI protected library files
+        if (!v3Global.opt.libCreate().empty()) {
+            if (v3Global.rootp()->delaySchedulerp()) {
+                v3warn(E_UNSUPPORTED, "Unsupported: --lib-create with --timing and delays");
+            }
+            V3ProtectLib::protect();
+            V3EmitV::emitvFiles();
+            V3EmitC::emitcFiles();
+        }
 
-    // Statistics
-    reportStatsIfEnabled();
+        if (v3Global.opt.stats()) V3Stats::statsStage("emit");
 
-    if (!v3Global.opt.lintOnly() && !v3Global.opt.xmlOnly() && !v3Global.opt.dpiHdrOnly()) {
-        size_t src_f_cnt = 0;
-        for (AstNode* nodep = v3Global.rootp()->filesp(); nodep; nodep = nodep->nextp()) {
-            if (const AstCFile* cfilep = VN_CAST(nodep, CFile))
-                src_f_cnt += cfilep->source() ? 1 : 0;
-        }
-        if (src_f_cnt >= V3EmitMk::PARALLEL_FILE_CNT_THRESHOLD) v3Global.useParallelBuild(true);
+        // Statistics
+        reportStatsIfEnabled();
 
-        // Makefile must be after all other emitters
-        if (v3Global.opt.main()) V3EmitCMain::emit();
-        if (v3Global.opt.cmake()) V3EmitCMake::emit();
-        if (v3Global.opt.gmake()) V3EmitMk::emitmk();
-    }
+        if (!v3Global.opt.lintOnly() && !v3Global.opt.xmlOnly() && !v3Global.opt.dpiHdrOnly()) {
+            size_t src_f_cnt = 0;
+            for (AstNode* nodep = v3Global.rootp()->filesp(); nodep; nodep = nodep->nextp()) {
+                if (const AstCFile* cfilep = VN_CAST(nodep, CFile))
+                    src_f_cnt += cfilep->source() ? 1 : 0;
+            }
+            if (src_f_cnt >= V3EmitMk::PARALLEL_FILE_CNT_THRESHOLD)
+                v3Global.useParallelBuild(true);
+
+            // Makefile must be after all other emitters
+            if (v3Global.opt.main()) V3EmitCMain::emit();
+            if (v3Global.opt.cmake()) V3EmitCMake::emit();
+            if (v3Global.opt.gmake()) V3EmitMk::emitmk();
+        }
+    }  // End: V3MtDisabledLockGuard{v3MtDisabledLock()};
```

Comment:
```suggestion
    }
```
End comments tend to get stale, so avoided (likewise elsewhere please).  If there's so much code it's hard to see where the block ends, the function's getting too large and should get split.